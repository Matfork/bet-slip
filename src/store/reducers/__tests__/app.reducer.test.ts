import { AppActionCreators } from '../../actions/action-creators/app.actionCreator';
import appReducer, { initialState } from '../app.reducer';

describe('App Reducer', () => {
  it('should match snapshot', () => {
    const state = appReducer(initialState, {
      type: 'other',
    });

    expect(state).toMatchSnapshot();
  });

  it('should perform IsLoading action', () => {
    const state = appReducer(initialState, {
      type: AppActionCreators.IsLoading,
    });
    expect(state).toMatchSnapshot();
  });

  it('should perform FinishedLoading action', () => {
    const state = appReducer(initialState, {
      type: AppActionCreators.FinishedLoading,
    });
    expect(state).toMatchSnapshot();
  });

  it('should perform SetError action', () => {
    const state = appReducer(initialState, {
      type: AppActionCreators.SetError,
    });
    expect(state).toMatchSnapshot();
  });

  it('should perform ClearError action', () => {
    const state = appReducer(initialState, {
      type: AppActionCreators.ClearError,
    });
    expect(state).toMatchSnapshot();
  });

  it('should perform ClearData action', () => {
    const state = appReducer(initialState, {
      type: AppActionCreators.ClearData,
    });
    expect(state).toMatchSnapshot();
  });
});
