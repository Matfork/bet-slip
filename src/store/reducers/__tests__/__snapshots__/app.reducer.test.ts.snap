// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Reducer should match snapshot 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;

exports[`App Reducer should perform ClearData action 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;

exports[`App Reducer should perform ClearError action 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;

exports[`App Reducer should perform FinishedLoading action 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;

exports[`App Reducer should perform IsLoading action 1`] = `
Object {
  "error": null,
  "isLoading": true,
}
`;

exports[`App Reducer should perform SetError action 1`] = `
Object {
  "error": undefined,
  "isLoading": false,
}
`;
